docs/develop/java/kafka.mdx

```diff
@@ -2,71 +2,30 @@
 sidebar_position: 10
 description: "Receive events from Kafka."
 ---
-
 import Admonition from '@theme/Admonition';
-import Tabs from '@theme/Tabs';
-import TabItem from '@theme/TabItem';
-
-# Kafka
-
-Restate can consume events directly from Kafka. To do that, you need to:
-
-* Develop an event handler, to consume events within your Restate service
-* Deploy Restate with the configuration to connect to a Kafka cluster
-* Wire up an event handler with a Kafka topic through a subscription
-
-## Event handler
-
-An event handlers is alike any other handler in your services/virtual objects, and profit from the same features, including durable execution, [service communication](/develop/java/service-communication) and [side effects](/develop/java/side-effects).
 
-**Handling keyed events within virtual objects**
+import {SubtleStep} from "../../../src/components/Stepper";
 
-You can handle events within virtual objects, where the key of the Kafka record matches the object key.
-
-Event handlers are no different from regular handlers, and the event payload will be (de)serialized as JSON:
-
-<Tabs groupId="sdk" queryString>
-<TabItem value="java" label="Java">
-
-```java
-CODE_LOAD::java/src/main/java/develop/MyKafkaVirtualObject.java
-```
+# Kafka
+You can invoke handlers via Kafka events, by doing the following:
 
-</TabItem>
-<TabItem value="kotlin" label="Kotlin">
+<SubtleStep stepLabel="1" title="Develop an event handler">
 
-```kotlin
-CODE_LOAD::kotlin/src/main/kotlin/develop/MyKafkaVirtualObject.kt
-```
+You can invoke any handler via Kafka events.
+The event payload will be (de)serialized as JSON.
```

Comment:
When sending to virtual objects the key must be valid UTF-8 String, this is really important to say because it could fail in unexpected ways otherwise.

> The event payload will be (de)serialized as JSON.

This is not really true, if you don't use the annotation based api you can get the raw payload, and once we get https://github.com/restatedev/sdk-java/issues/272 this will work with annotations too.
I would remove this sentence and rather replace it with a tooltip saying "if you use annotations it's always json, you can workaround with manual service definition, and in future we'll support this in annotations too"

Reply:
Let's document this once it works with annotations, because 90% of users will use annotations.

Reply:
Added it back as a details section